src/server/replica.cc

```diff
@@ -893,6 +893,7 @@ void Replica::RedisStreamAcksFb() {
   }
 }
 
+// TODO(kostasrim): Remove this on 20/6/2024
```

Comment:
I am totally confused. Should not you remove the replica side checks so that later you will be able to remove master side responses?
Where do you remove acl-check as written in the PR title?

Reply:
Sorry, my bad here. While I was making the changes I realized that the code I introduced above is enough to allow us a smooth transition (that means as long as two subsequent versions of df contain the code below we can remove both the acl-checks and be fine without the clients noticing.

```
      auto info_ptr = server_family_.GetReplicaInfo(dfly_cntx);
      if (info_ptr) {
        info_ptr->cntx.Cancel();
      }
      return;
```

However, I do agree we should remove this now. I will push an update soon.